### PR TITLE
Migrate public examples to v3

### PR DESCRIPTION
**Pull request by DzmitryFomchyn**
What has changed:
- Removed Drop In examples section
- Other examples are rewritten so that every time new route is requested instead of using hardcoded one (the coordinates for requested routes remain as they were in previous examples). This is due to custom routes are not supported in the Core Framework v3.
- Junction and signboard view examples have been removed because these 2 features are private and we provide them for selected customers only. These examples could work without feature flags for everyone on Nav SDK v2 because routes were hardcoded, but in the CF v3 custom routes are not supported now.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -73,8 +73,12 @@ android {
 
 dependencies {
     // Mapbox Navigation SDK
-    implementation "com.mapbox.navigation:android:2.10.1"
-    implementation "com.mapbox.navigation:ui-dropin:2.10.1"
+    implementation "com.mapbox.navigationcore:navigation:3.0.0"
+    implementation "com.mapbox.navigationcore:voice:3.0.0"
+    implementation "com.mapbox.navigationcore:copilot:3.0.0"
+    implementation "com.mapbox.navigationcore:ui-maps:3.0.0"
+    implementation "com.mapbox.navigationcore:ui-components:3.0.0"
```
Comment (LukasPaczos) [inline app/build.gradle:80]:
```suggestion
    implementation "com.mapbox.navigationcore:navigation:3.0.2"
    implementation "com.mapbox.navigationcore:voice:3.0.2"
    implementation "com.mapbox.navigationcore:copilot:3.0.2"
    implementation "com.mapbox.navigationcore:ui-maps:3.0.2"
    implementation "com.mapbox.navigationcore:ui-components:3.0.2"
```

Comment (LukasPaczos) [reply app/build.gradle:80]: Do we need all 5 deps? Why not `com.mapbox.navigationcore:navigation:3.0.2` only?

Comment (LukasPaczos) [reply app/build.gradle:80]: Hmm, I see that we'd also need to add `implementation "com.mapbox.navigationcore:ui-components:3.0.2"` which is not part of the primary artifact. This means that we're not including the UI components in the main artifact anymore, was that intentional? We're not mentioning this in any of the migration docs pages or documentation for UI components in https://docs.mapbox.com/android/navigation/guides/ui-components/.

File: app/src/main/java/com/mapbox/navigation/examples/standalone/fetchroute/FetchARouteActivity.kt [hunk 2]
```diff
@@ -12,26 +12,22 @@ import com.google.gson.JsonParser
...
-import com.mapbox.navigation.examples.R
 import com.mapbox.navigation.examples.databinding.MapboxActivityFetchARouteBinding
-import com.mapbox.navigation.examples.standalone.location.ShowCurrentLocationActivity
 
 /**
  * The example demonstrates given a pair of coordinates how to fetch a route.
  *
  * Before running the example make sure you do the following:
- * - Put your access_token in the correct place inside [app/src/main/res/values/mapbox_access_token.xml].
+ * - Put your access_token in the correct place inside [app-preview/src/main/res/values/mapbox_access_token.xml].
```
Comment (LukasPaczos) [inline app/src/main/java/com/mapbox/navigation/examples/standalone/fetchroute/FetchARouteActivity.kt:30]: Fix link.

File: app/src/main/java/com/mapbox/navigation/examples/standalone/fetchroute/FetchARouteActivity.kt [hunk 3]
```diff
@@ -40,7 +36,7 @@ import com.mapbox.navigation.examples.standalone.location.ShowCurrentLocationAct
  *   </resources>
  * - Add MAPBOX_DOWNLOADS_TOKEN to your USER_HOME»/.gradle/gradle.properties file.
  *   To find out how to get your MAPBOX_DOWNLOADS_TOKEN follow these steps.
- *   https://docs.mapbox.com/android/beta/navigation/guides/install/#configure-credentials
+ *   https://docs.mapbox.com/android/navigation/build-with-coreframework/installation/#configure-credentials
```
Comment (LukasPaczos) [inline app/src/main/java/com/mapbox/navigation/examples/standalone/fetchroute/FetchARouteActivity.kt:39]: Fix link.

File: app/src/main/java/com/mapbox/navigation/examples/standalone/location/ShowCurrentLocationActivity.kt [hunk 4]
```diff
@@ -1,33 +1,31 @@
...
 import com.mapbox.navigation.examples.databinding.MapboxActivityUserCurrentLocationBinding
-import com.mapbox.navigation.ui.maps.NavigationStyles
 import com.mapbox.navigation.ui.maps.location.NavigationLocationProvider
 
 /**
  * The example demonstrates how to listen to your own location updates and represent it on the map.
  *
  * Before running the example make sure you do the following:
- * - Put your access_token in the correct place inside [app/src/main/res/values/mapbox_access_token.xml].
+ * - Put your access_token in the correct place inside [app-preview/src/main/res/values/mapbox_access_token.xml].
```
Comment (LukasPaczos) [inline app/src/main/java/com/mapbox/navigation/examples/standalone/location/ShowCurrentLocationActivity.kt:28]: Fix link.

File: app/src/main/java/com/mapbox/navigation/examples/standalone/location/ShowCurrentLocationActivity.kt [hunk 5]
```diff
@@ -36,7 +34,7 @@ import com.mapbox.navigation.ui.maps.location.NavigationLocationProvider
  *   </resources>
  * - Add MAPBOX_DOWNLOADS_TOKEN to your USER_HOME»/.gradle/gradle.properties file.
  *   To find out how to get your MAPBOX_DOWNLOADS_TOKEN follow these steps.
- *   https://docs.mapbox.com/android/beta/navigation/guides/install/#configure-credentials
+ *   https://docs.mapbox.com/android/navigation/build-with-coreframework/installation/#configure-credentials
```
Comment (LukasPaczos) [inline app/src/main/java/com/mapbox/navigation/examples/standalone/location/ShowCurrentLocationActivity.kt:37]: Fix link.

File: app/src/main/java/com/mapbox/navigation/examples/standalone/turnbyturn/TurnByTurnExperienceActivity.kt [hunk 6]
```diff
@@ -562,45 +559,43 @@ class TurnByTurnExperienceActivity : AppCompatActivity() {
...
+                        Date().time.toDouble(),
+                        Point.fromLngLat(-122.39726512303575, 37.785128345296805)
+                    )
                 )
             )
-        )
-        mapboxReplayer.playFirstLocation()
-        mapboxReplayer.playbackSpeed(3.0)
+            playFirstLocation()
+            playbackSpeed(3.0)
```
Comment (LukasPaczos) [inline app/src/main/java/com/mapbox/navigation/examples/standalone/turnbyturn/TurnByTurnExperienceActivity.kt:592]: Remove the playback speed from this example.
```suggestion
```

**Review by LukasPaczos — COMMENTED** [review]
Could you also fix the installation link in the `README.md`?

**Comment by LukasPaczos** [comment]
Could you also add a note to README.md about where the v2 docs can be found?

File: app/src/main/java/com/mapbox/navigation/examples/standalone/turnbyturn/TurnByTurnExperienceActivity.kt
Comment (LukasPaczos) [inline app/src/main/java/com/mapbox/navigation/examples/standalone/turnbyturn/TurnByTurnExperienceActivity.kt:1]: The route arrows in this example are rendered on top of the puck, could you fix as well?

File: README.md [hunk 7]
```diff
@@ -12,7 +12,7 @@ The repository is a collection of examples showing off the [Mapbox Navigation An
 
 ## Installation
 
-Make sure you follow the steps under the [Installation](https://docs.mapbox.com/android/beta/navigation/guides/install/) guide. Once you have your **public and secret access tokens ready**, do the following:
+Make sure you follow the steps under the [Installation](https://docs.mapbox.com/android/navigation/build-with-coreframework/installation/) guide. Once you have your **public and secret access tokens ready**, do the following:
```
Comment (LukasPaczos) [inline README.md:15]:
```suggestion
Make sure you follow the steps under the [Installation](https://docs.mapbox.com/android/navigation/guides/installation/) guide. Once you have your **public and secret access tokens ready**, do the following:
```

File: README.md [hunk 8]
```diff
@@ -30,6 +30,11 @@ Make sure you follow the steps under the [Installation](https://docs.mapbox.com/
 6. Ensure that your secret token is added to [build.gradle](./build.gradle) or to global `gradle.properties` as described in the installation guide.
 7. Run the examples under `app`.
 
+## Previous versions
+
+If you're looking for code examples from previous versions of the Navigation SDK, you can access them through the following links:
+- [Nav SDK v2](https://docs.mapbox.com/android/navigation/v2/guides/): https://github.com/mapbox/mapbox-navigation-android-examples/tree/main-v2
```
Comment (LukasPaczos) [inline README.md:36]:
```suggestion
- [Navigation SDK v2](https://docs.mapbox.com/android/navigation/v2/guides/): https://github.com/mapbox/mapbox-navigation-android-examples/tree/main-v2
```

File: README.md [hunk 9]
```diff
@@ -30,6 +30,11 @@ Make sure you follow the steps under the [Installation](https://docs.mapbox.com/
...
+- [Nav SDK v1](https://docs.mapbox.com/android/legacy/navigation/guides/): https://github.com/mapbox/mapbox-navigation-android/tree/v1.6.0/examples/
```
Comment (LukasPaczos) [inline README.md:37]:
```suggestion
- [Navigation SDK v1](https://docs.mapbox.com/android/legacy/navigation/guides/): https://github.com/mapbox/mapbox-navigation-android/tree/v1.6.0/examples/
```

File: app/src/main/java/com/mapbox/navigation/examples/standalone/fetchroute/FetchARouteActivity.kt [hunk 10]
```diff
@@ -40,7 +37,7 @@ import com.mapbox.navigation.examples.standalone.location.ShowCurrentLocationAct
...
```
Comment (LukasPaczos) [inline app/src/main/java/com/mapbox/navigation/examples/standalone/fetchroute/FetchARouteActivity.kt:40]:
```suggestion
 *   https://docs.mapbox.com/android/navigation/guides/installation/#configure-credentials
```

File: app/src/main/java/com/mapbox/navigation/examples/standalone/location/ShowCurrentLocationActivity.kt (hunk 5 above)
Comment (LukasPaczos) [inline app/src/main/java/com/mapbox/navigation/examples/standalone/location/ShowCurrentLocationActivity.kt:37]:
```suggestion
 *   https://docs.mapbox.com/android/navigation/guides/installation/#configure-credentials
```

File: app/src/main/java/com/mapbox/navigation/examples/standalone/turnbyturn/TurnByTurnExperienceActivity.kt
Comment (DzmitryFomchyn) [reply app/src/main/java/com/mapbox/navigation/examples/standalone/turnbyturn/TurnByTurnExperienceActivity.kt:1]: Good catch, thanks! Fixed.

File: app/build.gradle (hunk 1 above)
Comment (DzmitryFomchyn) [reply app/build.gradle:80]: Discussed internally, that we're good to keep ui-components in separate from the main artefact. I'll also update docs!

File: app/build.gradle [hunk 11]
```diff
@@ -73,8 +73,12 @@ android {
 
 dependencies {
     // Mapbox Navigation SDK
-    implementation "com.mapbox.navigation:android:2.10.1"
-    implementation "com.mapbox.navigation:ui-dropin:2.10.1"
+    implementation "com.mapbox.navigationcore:navigation:3.0.2"
+    implementation "com.mapbox.navigationcore:voice:3.0.2"
+    implementation "com.mapbox.navigationcore:copilot:3.0.2"
+    implementation "com.mapbox.navigationcore:ui-maps:3.0.2"
```
Comment (LukasPaczos) [inline app/build.gradle:79]:
```suggestion
```

File: app-preview/src/main/java/com/mapbox/navigation/examples/preview/fetchroute/FetchARouteActivity.kt
Comment (LukasPaczos) [inline app-preview/src/main/java/com/mapbox/navigation/examples/preview/fetchroute/FetchARouteActivity.kt:1]: Why was this example removed?

File: app-preview/src/main/java/com/mapbox/navigation/examples/preview/location/ShowCurrentLocationActivity.kt
Comment (LukasPaczos) [inline app-preview/src/main/java/com/mapbox/navigation/examples/preview/location/ShowCurrentLocationActivity.kt:1]: Why was this example removed?

Comment (DzmitryFomchyn) [reply app-preview/src/main/java/com/mapbox/navigation/examples/preview/location/ShowCurrentLocationActivity.kt:1]: This example was in app-preview module and was moved to app.

File: app-preview/src/main/java/com/mapbox/navigation/examples/preview/fetchroute/FetchARouteActivity.kt
Comment (DzmitryFomchyn) [reply app-preview/src/main/java/com/mapbox/navigation/examples/preview/fetchroute/FetchARouteActivity.kt:1]: This example was in app-preview module and was moved to app.

Comment (LukasPaczos) [reply app-preview/src/main/java/com/mapbox/navigation/examples/preview/fetchroute/FetchARouteActivity.kt:1]: Gotcha :+1:

**Review by LukasPaczos — APPROVED** [review]
:ship:

**Pull request merged**